docs/architecture/decisions/0005-hide-e2e-related-content-from-users.md

```diff
@@ -0,0 +1,66 @@
+from creation, to publishing, through to use in a document (in both Whitehall and Mainstream) and the update of a value.
+
+In both of these circumstances, we do create a bit of noise for users:
+
+- For the GOV.UK tests, we have a block that we say users shouldn't use, and;
+- for the Content Modelling tests, we create new blocks for each test run.
+
+Usually this isn't too much of a problem because the Integration environment is cleaned up every Monday, and the staging
+environment is cleaned up every day.
+
+However, recently, a decision has been made to disable the Integration and Staging environment refreshes temporarily,
+which means there are a lot of stale blocks in the environments. As the Integration environment is going to be used
+shortly by real users, we need to cut down on the noise for users.
+
+This is complicated by the fact that (by design) it's not possible to delete a block from Content Block Manager via the
+UI. It is a developer task (see [Deleting a content block](https://github.com/alphagov/content-block-manager/blob/main/docs/deleting_a_content_block.md)), 
+but this would be a challenging (and risky) thing to automate as part of the E2E tests.
+
+## Decision
+
+We have decided that we should hide any blocks that are created by or for the End-to-End tests for real users. We will
+do this by adding a new field to the `Document` model, `hide_from_search_index`, which will be set to `true` for any 
+documents that are created by or for the End-to-End tests.
+
+In order for this to happen, we will need to add a comma seperated `E2E_EMAILS` environment variable to the application. 
+If any user with one of these email addresses creates a block, then we set the `hide_from_search_index` field to `true`.
+
+We did consider adding a new field to the `User` model, but decided against it as when the integration and staging 
+environments are refreshed, the `User` records are deleted and recreated, so we would need to update the `User` records
+for all users in the environments each time.
+
+Adding the field to the `Document` model also means that we don't need to add a join to the `User` model when querying 
+for documents that are created by or for the End-to-End tests.
+
+## Consequences
+
+We will need to add the `E2E_EMAILS` environment variable to the application via the GOV.UK Helm charts. These are not
+secrets as such, so can be added in plain text.
+
+Once this is done, there will still be some E2E blocks that do not have this boolean field set, so we will need to 
+manually update them. We will do this via a Rake task instead of on the console as this is safer and more testable. 
+We can run the task manually and then delete the code once it has been run in all environments.
+
+This also gives us the flexibility to run the Content Modelling E2E tests more often (such as on a schedule,
+or after a deployment to integration or staging), because the risk of adding noise to the application will be lowered.
```

Comment:
Do we also need to think about periodically purging the integration and staging environment of these unwanted content blocks? Or are we happy to have an every increasing volume of cruft in the databases?

In either case is there a case to be made that it would be simpler overall to deal with teardown of these tests as part of running the test harness? e.g. if each test run sends a `DELETE to /ops/content-block/{blockId}`. The code invoked could have guards in place e.g.:

- return if env is production
- return if document creator is not the E2E test user

Reply:
The problem we have is that these blocks don't just exist in our system - as soon as a block is published it makes a footprint in Publishing API and Content Store, as well as (if used) in other publishing apps. We do have the ability to delete content blocks from the app and Publishing API, but we can only do this if they're not used anywhere else within publishing.